src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt

```diff
@@ -156,6 +157,13 @@ class ReferralController(
     return DraftReferralDTO.from(updatedReferral)
   }
 
+  @PatchMapping("/draft-referral/{id}/desired-outcomes")
+  fun setDraftReferralDesiredOutcomes(authentication: JwtAuthenticationToken, @PathVariable id: UUID, @RequestBody request: SetDesiredOutcomesRequestDTO): DraftReferralDTO{
+    val referral = getDraftReferralForAuthenticatedUser(authentication, id)
+    val updatedReferral = referralService.updateDraftReferralDesiredOutcomes(referral, request.serviceCategoryId, request.desiredOutcomesIds)
+    return DraftReferralDTO.from(updatedReferral)
+  }
+
```

Comment:
I'm assuming here that we can allow the caller to remove all desired outcomes for a service category. 

I assume this will be required if the user deselects the service category from the make a referral - we will also need to remove any desired outcomes as well.

However the behaviour before was to prevent it being unset... there could be a check in the ReferralService to ensure that at least one desired outcome for all service categories is left. Do you think I should add this check?

Reply:
we need to explicitly stop unsetting of desiredoutcomes. previously this check handled this in `validateDraftReferralUpdate`

```
      if (it.isEmpty()) {
        errors.add(FieldError(field = "desiredOutcomesIds", error = Code.CANNOT_BE_EMPTY))
      }
      ```

Reply:
(talking to myself here)

.. we should make sure that we remove desiredOutcomes and complexityLevel for a given serviceCategoryId when it is removed via the updateDraftReferral route.

Reply:
not following the comment about removing them from updateDraftReferral.. aren't they already removed in this PR here https://github.com/ministryofjustice/hmpps-interventions-service/pull/253/files#diff-f7a6efe2090c4b294417026b860873e6c0fbc3f1b603412b45a294f1e44150c9L250 ?

i think we just need to check if the list of desiredoutcomes is empty in the new referral service method for updating the outcomes

Reply:
Added back in the check for empty list. 

Setting to empty list is still possible via UpdateDraftReferral when unselecting a service Category. This will be a new PR with a new ticket (still to be created).